jdaviz/core/template_mixin.py

```diff
@@ -1057,6 +1058,32 @@ def __init__(self, *args, **kwargs):
                                            allowed_type='spatial')
 
 
+class DatasetSpectralSubsetValidMixin(VuetifyTemplate, HubListener):
+    """
+    Adds a traitlet tracking whether self.dataset and self.spectral_subset
+    overlap in the spectral axis.
+
+    Note that if using in another method that is also observing dataset_selected
+    or spectral_subset_selected, that that method could be called before the traitlet
+    is updated.  In that case, call self._check_dataset_spectral_subset_valid() itself
+    directly.  The traitlet can still be used for any warning text in the plugin UI.
+    """
+    spectral_subset_valid = Bool(True).tag(sync=True)
+
+    @observe("dataset_selected", "spectral_subset_selected")
+    def _check_dataset_spectral_subset_valid(self, event={}):
+        # TODO: does this window not account for gaps?  Should we add the warning?
```

Comment:
Should probably open a follow-up issue for this TODO and put it in the comment?

Reply:
this TODO was not added in this PR (just moved).  There are lots of similar notes & caveats in model fitting, maybe we should open a general code-review/validation effort ticket?

Reply:
Ah, ok. There is a science validation ticket somewhere already, so I guess no need for new ticket.

Reply:
I'll remove this TODO now that #1947 is merged (which no longer passes window, but rather applies the mask)